gcloud/datastore/__init__.py

```diff
@@ -69,6 +69,50 @@
 _DATASET_ENV_VAR_NAME = 'GCLOUD_DATASET_ID'
 
 
+def _find_true_dataset_id(dataset_id, connection=None):
```

Comment:
I'd prefer they just use `s~foo`, but hopefully none of that will matter once the backend is updated.